package tvserver;

message tuner_tune {
    enum parameter_type_e {
        VCHANNEL = 0;
        CHANNELSCONF = 1;
    }

    message vchannel_m {
        optional uint32 vchannel = 1;
    }
    
    message channels_conf_record_m {
        optional string name = 1;
        optional uint32 frequency = 2;
        optional string modulation = 3;
        optional uint32 video_id = 4;
        optional uint32 audio_id = 5;
        optional uint32 program_id = 6;
    }

    message target_m {
        optional string host = 1;
        optional int32 port = 2;
    }
    
    required uint32 version = 1;
    optional string tuning_bigid = 2;

    optional parameter_type_e parameter_type = 3;
    optional vchannel_m vchannel = 4;
    optional channels_conf_record_m channelsconf_record = 5;
    optional target_m target = 6;
}

